core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/storage/ScanSourceFileResolver.scala

```diff
@@ -0,0 +1,37 @@
+package edu.uci.ics.texera.workflow.common.storage
+
+import java.nio.file.{Files, Path, Paths}
+import edu.uci.ics.amber.engine.common.storage.DatasetFileDocument
+import scala.util.{Failure, Success, Try}
+
+object ScanSourceFileResolver {
+
+  case class SourceFileNotFound(message: String) extends Exception(message)
```

Comment:
Remove source. It's even better to use an existing built in exception for such an error, if there is one.

Reply:
use FileNotFound exception from apache vfs